Sources/Apodini/Properties/Options/IdentifiyingType.swift

```diff
@@ -0,0 +1,30 @@
+//
+// Created by Andreas Bauer on 20.01.21.
+//
+
+/// Defines the identifying type passed to a `PathParameter`.
+/// The identifying type is required to conform to `Identifiable`,
+/// thus we save the `type` itself and its `Identifiable.ID` type.
+/// Such definitions are e.g. used to match property resolvers of Relationship
+/// definitions to their path parameter they actually resolve.
+public struct IdentifyingType: PropertyOption, Equatable {
+    /// The `Identifiable` type.
+    public let type: Any.Type
+    /// The `Identifiable.ID` type.
+    public let idType: Any.Type
+
+    public static func == (lhs: IdentifyingType, rhs: IdentifyingType) -> Bool {
+        lhs.type == rhs.type && lhs.idType == rhs.idType
+    }
+}
```

Comment:
I would suggest to create an internal initializer that can only be created with an `Identifiable` type. Otherwise it would be hard to enforce that the `type` and `idType` match.
In addition I think that comparing the `type` should be sufficient for the `Equatable` implementation if we make sure that `idType` always matches the initial `Identifiable.ID` type.
```suggestion
public struct IdentifyingType: PropertyOption, Equatable {
    /// The `Identifiable` type.
    public let type: Any.Type
    /// The `Identifiable.ID` type.
    public let idType: Any.Type

    public static func == (lhs: IdentifyingType, rhs: IdentifyingType) -> Bool {
        lhs.type == rhs.type
    }
    
    init<Type: Identifiable>(identifying type: Type.Type = Type.self) {
        self.type = type
        self.idType = type.ID
    }
}
```

Reply:
I would aim to keep `idType` in the equality check. After all, this is what caught the issue with the AssociatedTypeRequirementsKit and its type visitor. But yes, the initializer is a good idea.

Reply:
Solved via 5ac7d7df3c2c1de6c3fd59031fa9f758d6822218